src/components/main/Main.tsx

```diff
@@ -1,29 +1,43 @@
 import styles from './Main.css';
 import commonStyle from '../../styles/commonStyles.css';
 
-import React, { useContext, FC, useEffect, useState } from 'react';
+import React, { useContext, FC, useEffect, useState, useRef } from 'react';
+import { useParams } from 'react-router-dom';
 import { positionContext } from 'context/positionContext';
 import { getCity } from 'services/getCity';
 import { getWeather } from 'services/getWeather';
 import { CurrentWeatherType, HourlyWeatherType, DailyWeatherType } from 'types/weatherTypes';
 import { LocationInfoType } from 'types/cityInfoType';
-import { defaultCurrentWeather, defaultLocationInfo } from './defaultStates';
+import { defaultLocationInfo } from './defaultStates';
 import MainCard from './MainCard';
 import MainHourlyCard from './MainHourlyCard';
 import MainDailyCard from './MainDailyCard';
 
+type PositionRef = {
+  lon: number | string;
+  lat: number | string;
+};
+
 const Main: FC = () => {
   const [location, setLocation] = useState<LocationInfoType>(defaultLocationInfo);
   const [currentWeather, setCurrentWeather] = useState<CurrentWeatherType | undefined>(undefined);
   const [hourlyWeather, setHourlyWeather] = useState<HourlyWeatherType[]>([]);
   const [dailyWeather, setDailyWeather] = useState<DailyWeatherType[]>([]);
   const [selectDays, setSelectDays] = useState<string>('');
 
+  const { coordinates } = useParams();
+  const [paramLongitude, paramLatitude] = (coordinates || '').split(',');
+
   const {
     state: { position, positionError }
   } = useContext(positionContext);
 
-  async function fetchData(lon: number, lat: number) {
+  const { current: positionRef } = useRef<PositionRef>({ lon: 0, lat: 0 });
```

Comment:
what is the reason to use ref instead of simple state here?

Reply:
I don't see the reason to use state for this data. 
I get user location from context and selected position (new city form favorite or search) form URL. I'll get new data from URL on every render. I think I don't need useState for result info about position. 
I use useRef for save info in variable.

Reply:
no need to use ref for it, use plain variable.

Reply:
when I work with timerId you said me don't use plain  variable.
in which case I need to use ref, in which  plain  variable?

Reply:
if not touching cases related to DOM manipulation, refs should be used only if you need variable inside your component, that should be stable between re-renders (e.g, another solutions, like plain variable and state, doesn't work). in case with timerId you mentioned i advised to use ref because: 
1. previously you had `let` variable outside of component body and in case you'd think in future about moving this functionality to custom hook and reuse in few places, it could cause bugs
2. just to try use ref in "classic" case where it can be useful